lib/Aggregator.php

```diff
@@ -491,23 +500,46 @@ protected function getEntitiesDescriptor()
         // add RegistrationInfo extension if enabled
         if (!empty($this->regInfo)) {
             $ri = new RegistrationInfo();
-            $ri->registrationInstant = $now;
+            $ri->setRegistrationInstant($now);
             foreach ($this->regInfo as $riName => $riValues) {
                 switch ($riName) {
                     case 'authority':
-                        $ri->registrationAuthority = $riValues;
+                        $ri->setRegistrationAuthority($riValues);
                         break;
                     case 'instant':
-                        $ri->registrationInstant = Utils::xsDateTimeToTimestamp($riValues);
+                        $ri->setRegistrationInstant(Utils::xsDateTimeToTimestamp($riValues));
                         break;
                     case 'policies':
-                        $ri->RegistrationPolicy = $riValues;
+                        $ri->setRegistrationPolicy($riValues);
                         break;
                 }
             }
             $ret->Extensions[] = $ri;
         }
 
+        // add PublicationInfo extension if enabled
+        if (!empty($this->pubInfo)) {
```

Comment:
According to the spec, the `publisher` property of the `mdrpi:PublicationInfo` element is required. As such, I think we should check if the configuration contains it so that we don't generate an XML that's not conformant to the spec. That should probably include a check here (throwing an exception if the config option is missing) and some comment in the configuration to make it clear that it's required.

Reply:
The SAML2-library takes care of that [here](https://github.com/simplesamlphp/saml2/blob/release-3.x/src/SAML2/XML/mdrpi/PublicationInfo.php#L59).

Reply:
Actually, that code is only run when the `PublicationInfo` object is built from an existing XML document, but in this case we're building an empty one and filling in the details. As far as I can see, the `toXML()` method does not check it is set to something. The only thing that would avoid a malformed element is [this](https://github.com/simplesamlphp/saml2/blob/release-3.x/src/SAML2/XML/mdrpi/PublicationInfo.php#L123), and I'm not even sure it would throw an exception. If it does, the message would be a bit misleading and hard to relate to a misconfiguration, IMHO.

Reply:
Ohh, i posted the wrong line.. I meant this one here: https://github.com/simplesamlphp/saml2/blob/release-3.x/src/SAML2/XML/mdrpi/PublicationInfo.php#L171

A null will definitely throw an exception here

Reply:
Ah, right, I missed that one! Still, I think it's worth adding a check, so that we can throw a meaningful exception that points towards the actual error...

Reply:
Should we really raise an exception or log a warning and silently continue without PublicationInfo added?

Reply:
Uhm, that's a good question. If we continue, an exception will be thrown eventually, right? So maybe it's better to throw an error early if we know we'll stop later anyway...

Reply:
No, I mean, if we miss a publisher, we could still continue operation without adding a PublisherInfo to the metadata..

```
if (!empty($this->pubInfo)) {
  if (isset($this->pubInfo['publisher'])) {
    // do magic
  } else {
    // logging a warning
    Logger::warning('Missing publisher;  PubisherInfo not set in metadata.');
    // or rather raise an exception and stop doing anything at all
  }
}
```

Reply:
Ah, I see, sorry!

I'm not sure... In the end, if you configured it and the configuration is not empty, that's because you explicitly want it to be in place. On the other hand, it's a bit harsh to just drop metadata generation due to a failure that we can recover from...

Reply:
Let's just say that this module could use improvements in many ways... Not sure we should all make that part of this PR.. The way it is now, an exception will be thrown by the saml2-library and personally I'm fine with that

Reply:
Ok, let's leave it like that then and we'll try to enhance the module at a later time!